infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -418,27 +338,67 @@ private void handleTxToken(CorfuPayloadMsg<TokenRequest> msg,
-            Token token = new Token(Address.ABORTED, serverEpoch);
-            r.sendResponse(ctx, msg, CorfuMsgType.TOKEN_RES.payloadMsg(new TokenResponse(tokenType,
-                    conflictKey.get(), token, Collections.emptyMap())));
+        // The sequencer is a failover sequencer and we have no information,
+        // so the request fails.
+        if (txSnapshotTimestamp < globalLogStart.get() - 1) {
+            log.debug("handleTxToken: ABORT[{}] snapshot-ts[{}] failover-ts[{}]",
+                    txSnapshotTimestamp, globalLogStart.get());
+            r.sendResponse(ctx, msg, CorfuMsgType.TOKEN_RES.payloadMsg(
+                    new TokenResponse(TokenType.TX_ABORT_NEWSEQ, Address.ABORTED, serverEpoch)));
             return;
         }
 
+        // Iterate over all of the streams in the conflict set
+        for (Map.Entry<UUID, Set<Integer>> entry : txInfo.getConflictSet().entrySet()) {
+            final Set<Integer> conflictParamSet = entry.getValue();
+            final UUID streamId = entry.getKey();
+
+            // if conflict-parameters are present, check for conflict
+            // based on conflict-parameter updates
+            if (conflictParamSet != null && conflictParamSet.size() > 0) {
+                // for each key pair, check for conflict;
+                // if not present, check against the wildcard
+                conflictParamSet.forEach(conflictParam -> {
+                    final int conflictKeyHash = getConflictHashCode(streamId, conflictParam);
+                    final Long conflictTail = conflictToGlobalTailCache
+                            .getIfPresent(conflictKeyHash);
+
+                    if (conflictTail != null
+                            && (conflictTail > txSnapshotTimestamp
+                            || maxConflictWildcard > txSnapshotTimestamp)) {
+                        log.debug("handleTxToken: ABORT[{}] conflict-key[{}]({}ts={})", txInfo,
+                                conflictParam,
+                                maxConflictWildcard > txSnapshotTimestamp ? "WILDCARD " : "",
+                                conflictTail);
+                        r.sendResponse(ctx, msg,
+                                CorfuMsgType.TOKEN_RES.payloadMsg(
+                                        new TokenResponse(TokenType.TX_ABORT_CONFLICT_KEY,
+                                            conflictParam, conflictTail, serverEpoch)));
+                        return;
+                    }
+                    log.trace("handleTxToken: OK[{}] conflict-key[{}](ts={})", txInfo,
+                            conflictParam, conflictTail);
+                });
+            } else { // otherwise, check for conflict based on streams updates
+                final Long streamTail = streamTailToGlobalTailMap.get(streamId);
+                if (streamTail > txSnapshotTimestamp) {
+                    log.debug("handleTxToken: ABORT[{}] conflict-stream[{}](ts={})",
```

Comment:
That'll spray a lot of output on your logging, and it seems like you are logging debug level, so I think we should leave as is.